include/openthread/instance.h

```diff
@@ -150,6 +150,7 @@ enum
     OT_CHANGED_THREAD_BACKBONE_ROUTER_STATE = 1 << 25, ///< Backbone Router state changed
     OT_CHANGED_THREAD_BACKBONE_ROUTER_LOCAL = 1 << 26, ///< Local Backbone Router configuration changed
     OT_CHANGED_JOINER_STATE                 = 1 << 27, ///< Joiner state changed
+    OT_CHANGED_THREAD_DOMAIN_NAME           = 1 << 29, ///< Thread Domain Name changed.
```

Comment:
note: Bit 28 is used for Domain Prefix change, so here uses bit 29.

Reply:
Do we need to notify the user if/when Domain Name changed? 

At the moment, if I understand correctly  the only way to change Domain Name is through the public `otSetDomainName` API. Will there be other ways that core itself would change the Domain Name (like from dataset or network data or ...?)

Reply:
Yes, currently it could only be changed via `otSetDomainName`, I am fine to remove it as the notification seems not so mandatory so far.